src/foam/u2/Scroller.js

```diff
@@ -0,0 +1,173 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'Scroller',
+  extends: 'foam.u2.Element',
+
+  /**
+   * Wraps a TableView or similar and adds a ScrollView to it.
+   * Configure the table view with the tableView property, and the scrollbar
+   * with the scrollView property.
+   * Set data to the DAO, and the tableView will receive that DAO with skip
+   * applied correctly.
+   *
+   * This view needs to know the size of its container, in order to size the
+   * tableView and scrollbar accordingly. Therefore it needs a fixed row height
+   * to use for the table. Set rowHeight to the number of pixels per row.
+   */
+  requires: [
+    'foam.graphics.Canvas',
+    'foam.graphics.ScrollCView',
+    'foam.mlang.sink.Count',
+    'foam.u2.TableView',
+    'foam.u2.ViewSpec'
+  ],
+
+  imports: [
+    'window'
+  ],
+
+  properties: [
+    {
+      name: 'of',
+      expression: function(data) { return data.of; }
+    },
+    {
+      name: 'data',
+      required: true,
+      postSet: function(old, nu) {
+        // TODO(braden): When a better way of reacting to DAOs becomes
+        // available, use it here.
+        if ( old ) old.on.unsub(this.onDAOUpdate);
+        if ( nu ) nu.on.sub(this.onDAOUpdate);
+        this.onDAOUpdate();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'rowHeight',
+      value: 36
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'tableView',
+      value: { class: 'foam.u2.TableView' }
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'scrollView',
+      value: { class: 'foam.graphics.ScrollCView' }
+    },
+    {
+      /** The actual TableView instance. */
+      name: 'table'
+    },
+    {
+      /** The actual ScrollCView instance. */
+      name: 'scrollBar'
+    },
+    'scrollValue_',
+    'scrollHeight_',
+    'scrollExtent_',
+    'pointer'
+  ],
+
+  methods: [
+    function initE() {
+      var self = this;
+      var canvas = this.Canvas.create(null, this);
+      canvas.attrs({ height: this.scrollHeight_$ });
+
+      this.cssClass(this.myCls());
+
+      this.table = this.start().cssClass(this.myCls('container'))
+          .start(this.tableView, {
+            of: this.of,
+            data$: this.slot(function(dao, extent, value) {
+              return dao.limit(extent).skip(value);
+            }, this.data$, this.scrollExtent_$, this.scrollValue_$)
+          });
+      this.table.end().end();
+
+      // TODO: Fix this when we have the improved ViewSpec handling, which
+      // should allow specifying the context (parent) for the spec.
+      this.startContext({ pointer: canvas.__subContext__.pointer });
```

Comment:
Yes, you're looking at the old commits, I think. It has been changed and uploaded.